### PR TITLE
Fix running KubeletManagedEtcHosts conformance tests from a remote de…

### PR DESCRIPTION
**Pull request by dims**
…v environment

**What this PR does / why we need it**:
In ba20e63446815f8bcab4fc4c6e07eab2e2a9e121, we did not account for the fact that the e2e tests may be triggered from a remote dev environment in which case the generated etc/hosts file will not be in the node(s). So let's check if the local /etc/hosts has the managed header and avoid creating the temporary file.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Please see note from @karataliu here:
https://github.com/kubernetes/kubernetes/pull/60627#issuecomment-374554960

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
/hold

( for feedback from @karataliu )

**Comment by karataliu** [comment]
Added tracking issue #61451

File: test/e2e/common/kubelet_etc_hosts.go [hunk 1]
```diff
@@ -81,24 +83,32 @@ func (config *KubeletManagedHostConfig) verifyEtcHosts() {
 }
 
 func (config *KubeletManagedHostConfig) setup() {
-	etcHostContents := `127.0.0.1	localhost
+	var err error
+
+	// create a temporary /etc/hosts file if local /etc/hosts has the managed host header
+	hostsFileContent, err := ioutil.ReadFile("/etc/hosts")
+	if err != nil && bytes.HasPrefix(hostsFileContent, []byte(etcHostsPartialContent)) {
```
Comment (karataliu) [inline test/e2e/common/kubelet_etc_hosts.go]: if err!=nil, hostsFileContent would be invalid?

Comment (dims) [reply test/e2e/common/kubelet_etc_hosts.go]: oops!

**Comment by dims** [comment]
/close 

in favor of 61613

**Pull request closed without merging**